ННКСС (Научно Наблюдательная Космическая Станция Синдиката) "Тайпан"

Furukia opened the pull request:
Синдибаза "Тайпан" и всё что было сделано с ней или для неё.

<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. Document every changes or this may delay review or even discourage maintainers from merging your PR! -->

Добавлена локация ННКСС (Научно Наблюдательная Космическая Станция Синдиката) "Тайпан"

•	Спавнится в космосе ВСЕГДА, как и астероид синдиката из-за своих объёмов вместе с метеоритами занимает почти весь уровень, но всё ещё позволяет другим случайным объектам сравнится в космосе на одном с ней уровне как обычно. 

На станции 11 спаунов

•	Офицер Телекоммуникаций - По совместительству один из двух руководящих ролей. Его задача наблюдать за станцией НТ через телекамеры. Руководить станцией в случае любого рода боевой ситуации и охранять секретные документы синдиката в хранилище и оружейную. Он единственный имеет доступ к вышеназванным зонам и Телепортной.
•	Директор Исследований - Второй глава на станции. Его задача похожа на задачу стандартного директора исследований НТ. Следить чтобы учёные делали свою работу и не взорвали свою же станцию. Так же на его плечи ложится задача сделать, что-либо для синдиката, если они решат заказать у станции что-нибудь. Единственный обладает доступом к серверу, удобно расположенному прямо под его кабинетом.
•	5 Слотов для Учёных - Они обладают доступом в рнд, медбей, вирусологию, ботанику и в инженерию (Вдруг захотят поэкспериментировать с газами). Их задача это любого рода исследования био и химоружия, генов и так далее. Им нельзя строить ИИ! Им можно строить боргов только с разрешения администрации! Возможно, если того захочет народ, я пороюсь в коде и постараюсь создать версию робомозга и мми для боргов, которые не будут привязываться к консоли боргов НТ. Но пока так!
•	2 Слота для Грузчиков - Они обладают доступом в первую очередь к карго. Их работа покупка, одобрение/отказ и продажа товаров через телепады в карго. А также доставка этих товаров тем, кто их закажет. Их карго отличается от НТ, и торговля ведётся на "Чёрном рынке" за кредиты. Так же у них немного отличается список товаров, и некоторые товары являются "краденными", что значит ящики могут иметь замок с особым доступом. Нет, Синдикат не просит прощения за неудобства.
•	1 Слот для Повара - Он обладает доступом на кухню, в бар и в ботанику. Работает, так сказать, за троих. Но ничего не мешает ему попросить учёных, которые увлеклись ботаникой вырастить чего-нибудь и для него.
•	1 Слот для Атмос Инженера - Он, так сказать, совмещает в себе работу атмосферника и инженера. Обладает доступом в Инженерный. Его задача настроить стоящий в атмосе Термоэлектрогенератор и чинить станцию в случае любого рода поломок или неизбежных взрывов.

**Считается, что все работники станции обучены обращению с оружием!
Секретность и целостность этой станции является приоритетом для агентов станции. Это значит, что любого рода помощь агентам и особенно не агентам ставящая под угрозу вашу секретность запрещена без одобрения администрации!**

Для карго реализованы специальные синдипады и консоль для взаимодействия с чёрным рынком! Работают с помощью редспейса! Игнорируют любого рода блокировки телепортации! 

•	Стандартно синдипады телепортируют только посылки! Трупы и живые организмы они не станут телепортировать. Админ может изменить это поставив флаг allow_humans на TRUE у телепада отправляющего посылки.
•	Синдипады используют систему ID адресов и работают в одном из двух режимов (Получение посылок и их отправка)
•	Отвёртка откроет панель телепада, открытая панель позволяет настраивать их другими инструментами или разобрать их с помощью лома.
•	Настройка через кусачки приведёт к попытке привязать синдипад к консоли заказов
•	Настройка через мультитул переключит режим работы синдипада и попросит вас ввести ID для этого телепада или ID телепада "цели" для отправки посылок.
•	Если синдипад не привязан к консоли он работает через прямую активацию по клику на него. Иначе при попытке клика он потребует применять консоль для его активации
•	Синдипады состоят из следующих компонентов:
o	Плата Синдипада
o	1 Телекристалл
o	Конденсатор
o	Манипулятор
o	1 Кусок Кабеля
•	Синдипады могут быть улучшены стандартной заменой компонентов, что сократит время для их активации и их кулдаун, как личный, так и в консоли!

Консоль заказов.

•	В целом работает почти так же, как консоль в обычном карго. Но использует телепады вместо шатла и кредиты вместо очков. 
•	Для консоли был создан уникальный интерфейс (Все те файлы в папочке TGUI важны!) Если с ним возникнут проблемы пожалуйста сообщите мне.
•	Кредиты можно внести в консоль простым кликом по ней деньгами. Их так же можно снять с консоли с помощью специальной кнопки. У админов есть секретная кнопка для их добавления (или уменьшения если прописать там минус перед суммой). Кнопка добавлена дабы админы могли заплатить за заказанные ими товары у станции или просто поощрить за работу. 
•	Список товаров в консоли отличается наличием множества контрабанды и другими ценами за товары в кредитах вместо очков. Для справедливых цен мне необходимы отзывы игроков, поэтому все цены что вы видите там сейчас, подлежат возможным изменениям, как и сам список товаров.

Другие мелочи:

•	Для карго был создан специальный синдиящик, который нельзя взломать емагом и который не удаляется вместе с содержимым при телепорте, а отправляется прямо на телепад к админам на синдибазе в снегах. (Ах да, поставила там два телепада. Если, что передвинете их или ещё чего)
•	Для станции так же было реализовано несколько уникальных синдикатовских видов дверей. Они все прочнее своих нт аналогов на 50 единиц и делаются из пластитана. Но в связи с отсутствием многих зон, которые есть у нт, были реализованы аналоги не для всех дверей. Возможно, потом я заспрайчу и добавлю оставшиеся аналоги для любителей строить такие двери на обычной станции.
•	Были добавлены уникальные доступы для профессий на станции. А также все эти доступы были учтены и добавлены в карту админского офицера синдиката.
•	Немного был переделан код астероида синдиката. Если точнее, то код компьютера телепортера. Переделанный код был скопирован с официального парадайза, и он просто убирает ограничение, из-за которого компьютер не работал в других зонах, кроме зоны астероида.
•	Немного переделан код серверов рнд, для того чтобы создать возможность синдикату иметь независимую сеть в которой не видно серверов НТ, так же как НТ не видит сервера синдиката. Это всё ещё не мешает големам воровать уровни исследований у НТ и реализовано через простой флаг для проверок.
•	Спавнеры после появления игрока, заменяются на крио капсулу со скином синдиката.
•	Автомат "Синди Робо-ДеЛюкс!" стоящий около пыточной и содержащий детали для боргов и 2 синдикатских мми. 
•	Логотип синдиката в центре базы.
•      Уникальный Генерал Синди (Грифски с личностью) Умеет отличать синдикатовцев от прочих личностей. Делает он это смотря на две вещи. Фракцию "syndicate" встроенную в персонажа и на наличие карты агента. Если нет хотя бы одного из этого списка, он начинает бить. Об "арестах" он кричит в канал синдиката.



## Why It's Good For The Game
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Желаемая игроками и прошедшая админское голосование гост роль станции синдиката. Уникальный контент которому (я надеюсь) обзавидуется оригинальный Paradise!

## Images of changes

ННКСС "Тайпан"
<!-- If you did not make a map or sprite edit, you may delete this section. You may include a gif of your feature if you want -->
![syndiebase](https://user-images.githubusercontent.com/67807868/143624011-6193884c-2d03-4968-a976-70fa5b4a44a9.png)

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Furukia committed:
Тайпан

Добавление недостающих файлов в файл компилятор и фикс рантаймов с трубами от Wojciech Łobaczewski

Ещё включила выключенные скрабберы и добавила одной мусорке недостающую трубу.

Синдибаза и всё что было сделано с ней или для неё.

Furukia committed:
Убрала лишний флаг синдипадов.

Убрала найденный неиспользуемый флаг у синдипадов. Процесс был реализован иначе первоначальной идеи поэтому флаг стал бесполезен.

Furukia committed:
СиндиГрифски, исправление проблем с новыми доступами и мышками, нормальная зарплата в карго и клонёрка в меде.

Добавлена версия грифски которая умеет отличать обычных людей от синдикатовцев и давать по лицу первым.

Всем мышам станции был выдан параметр фракции "syndicate" чтобы турели по ним не стреляли. Добро пожаловать в синдикат! Так же был поправлен доступ на паре ящиков.

Карго наконец то начали платить за работу в 100 раз больше чем раньше за технологии от рнд!

Офицеры синдиката больше не испытывают проблем с картами без новых доступов.

Добавлена клонёрка. Синдикат не видел смысл тратиться на отдельное помещение для неё, поэтому она просто стоит в зале медпункта!

Update griefsky.dm

Furukia committed:
Сделала файр алярмы на Тайпане рабочими плюс мелкие изменения на карте

Сделала файр алярмы на Тайпане рабочими.
Изменения в расположении зон для логичной работы некоторых файрлоков.
Добавила консоль боргов в кабинет рд (она видит только боргов на своём z уровне)
Light replacer добавлен в коморку для уборки вместе с janitorial cart
Поправлен товар в Синди Робо-ДеЛюкс

Furukia committed:
Создала систему для спавна синдиборгов через синди-робомозг

При использовании этого мозга вместо стандартных модулей доступны на выбор модули Саботажника, Синдимедборга и Штурмовой. Этот мозг не допускает установку стандартных модулей даже через модуль reset.

Сам мозг можно найти в товарах в карго на Тайпане.

Так же сделала возможность саботажа каркаса для боргов емагом полезной. Теперь если каркас емагнуть до вставления в него мозга, даже при наличии ИИ, привязки у борга не произойдёт во время его создания. (Полезно для создания обычных нт боргов на тайпане. Можно даже законы не менять. Все законы подходят для работы и на этой станции)

Furukia committed:
Правки для Синди Грифски, изменение цены плазмы в карго и незначительные правки карты

Грифски поумнел ещё немного и теперь умеет кричать на НТ свиней и правильно определять не синдикатовцев чтобы бить их. Он смотрит на наличие карты агента или встроенной в код перса faction = "syndicate" В обоих случаях он не тронет, но если есть только карта, её лучше не снимать.

Цена плазмы в карго удвоена. 100 плазмы = 10000 кредитов

Незначительные изменения в карте, добавление пары сейфов с секретиками, слизней в заброшенную зону, на случай если станции захочется построить своё ксено. Так же добавлено пару синдипда для глав, украшений на полу в рнд и пару стоек под мусором в техах.

Добавила Desert Eagle (не золотой) в товары карго, как и патроны к нему.

Furukia committed:
Добавила СиндиНаномед.

Синди наномед это просто рескин наномеда выполненный в красно-чёрном стиле маленького настенного синдимеда. По товарам там тоже самое. но в меде на карте расположены два наномеда у которых есть премиум товар в виде исцеляющего луча, как у медборгов синдиката.

Furukia committed:
Синди двери, синди шкафы, фикс багов и пара изменений на карте

* Добавлены ещё несколько синдидверей, в этот раз заспрайченных Элл Гууд-ом. И рецепты для них.
* Новые синди-шкафы. Так же от Элл Гууда.
* Ребаланс формулы высчитывания кулдауна синдипадов в зависимости от компонентов.
* Теперь в карго содержимое обычных ящиков не телепортируется, а удаляется так же как и ящики, дабы не засорять z2. (Синдиящики всё так же телепортируются вместо простой продажи, вместе со всем содержимым.)

* Исправлен баг из-за которого можно было строить бесконечное количество Компьютеров синдикарго из одной платы.
* Фикс бага из-за которого телепады в режиме отправки после вывода сообщения о невозможности отправки, производили телепортацию!(Чего они делать были не должны)
* Фикс уничтожения людей синдипадом если производить телепорт человека через консоль при включённом флаге allow_humans
* Фикс бага на бесконечные деньги в карго консоли. Теперь при выдаче денег есть проверка на минус и плавающую точку. (За нахождение бага, отдельное спасибо He11Force-у)

Изменения на карте в виде:
* Расстановки новых ящиков
* Добавления ботаникам их устройства для дисков и самих дисков
* Выдачи всем дверям нормальных названий соответственно их зоне
* Замены всех обычных поясов на хамелионские пояса, по запросам тестеров
* Добавления пары снеговиков и снежного байка на снежный астероид не далеко от базы
* Добавления в карго ставней для окон и стойки. (Дабы закрыть окна от туррелей, что стреляют по прибывающим в карго коровам)
* Добавления Анализатора на стол в рнд (Для начальных исследований)
* Добавления пары новых точек спавна лута в техах
* Небольшой перестановки у рд
* Добавления интеркома в телепортерную
* Добавления синдикатского гостевого терминала. Позволяет создать гостевой пропуск с Тайпанскими доступами. Находится в кабинете Офицера Телекомов.

Furukia committed:
Ночное освещение, логи консоли, исправление опечатки и фракции эскимосов с астероида.

* Исправлена дурацкая опечатка в консоли карго
* Добавлено логирование снятия и добавления денег в консоль. (Логи консоли, не админские)
* Небольшие изменения кода ночного освещения, чтобы оно учитывало апк на синдистанции.
* Поправка фракций эскимосов на снежном астероиде, чтобы они не били снеговиков соседствующих с ними.

Furukia committed:
Правки багов и работы синдипадов.

*Добавление подвидов синдипадов для удобного маппинга
*Автоматическое установка нужного ID при привязке к консоли в зависимости от типа (отправка/получение)

Furukia committed:
Фиксы багов, добавление поварского CQC повару и т.д.

* Пару изменений в доступе и/или типе некоторых дверей
* Вернула ящик с инструментами в гидропонику
* Накидала ещё кучку кошельков в разные места на станции
* Накидала парочку зарядников для энергетического оружия
* Пофиксила неразрушаемость machine frame-ов в техах
* Пофиксила разрушаемость невидимого хранилища данных карго
* Оставила небольшую отсылочку на своих персонажей в виде помады
* Чуть чуть дополнила текст на бумажке в карго дабы уменьшить количество не понимающих какие ящики куда слать людей
* Добавила privacy shutters в рнд
* Добавила зеркала в генетику (ибо учитывая особенности создания персонажей под эту роль, стабильный мутаген или игры с UE блоками могут сбросить причёску и усы персонажа. Так хоть вернуть себе быстро)
* Теперь введённое имя персов с базы сохраняется в днк, так что стабильный мутаген с вашей кровью не превратит вас внезапно в Иванова Ивановича
* Пофиксила белую кожу через костяк в виде обновления днк (Как если применить стабильный мутаген, но без урона)
* У синди повара теперь есть поварское CQC
* У телекомщика теперь в сумке спавнится бумажка со списком кодовых слов предателей
* У всех кроме телекомщика теперь наушники привязаны только к частоте предателей (Теперь не будет случайного спама на обычную общую частоту НТ)

Furukia committed:
Синди туррели теперь обращают внимание на карты агента + Разные типы крови для Тайпановцев

* Синди туррели теперь обращают внимание на карту агента (Так как использование только прописанной в самих мобах фракции не очень надёжно. Например она слетает при клонировании и внезапно свои туррели начинают агрессировать на вас.)
* Теперь агентам при спавне выдаётся рандомный тип крови. До этого у всех было "A+"

Furukia committed:
Змея телекоммщика Руж и защита от грифа мышей

* Добавлен уникальный питомец синдиката, трёхголовая змея Руж
* Добавлен код позволяющий ей носить шляпки. На данный момент она умеет носить только беретик ХоСа, но добавить новые шляпы не должно быть сложно если уже есть спрайты.
* Собственно добавила Руж на карту и кроватку для неё.
* Мыши при спавне на синдибазе получают фракцию синдиката. Это нужно чтобы мыши не могли подойти к окнам на базе, заагрить туррели и устроить разгермы или того хуже взорвать фуелтанки у окон.

Vallat commented on code/controllers/subsystem/nightshift.dm:
```diff
@@ -57,6 +57,7 @@ SUBSYSTEM_DEF(nightshift)
 	for(var/A in GLOB.apcs)
 		var/obj/machinery/power/apc/APC = A
-		if(is_station_level(APC.z))
+		var/area/APCAREA = get_area(APC)
+		if(is_station_level(APC.z) || istype(APCAREA, /area/syndicate/unpowered/syndicate_space_base))
```
Правда ли это хорошая идея синхронизировать тип освещения (ночное/дневное) между синдистанцией и обычной? Не дать ли синдистанции по умолчанию дневной свет и пусть они сами переключают APCшки как хотят?

Furukia replied:
Это очень важно для уюта и атмосферы. Всё таки это локация на которой свой экипаж играет свою смену. А ходить и тыкать каждую апк запарно. Делать какую-нибудь кнопку для переключения всех апк на синдистанции в ночной режим, тоже такое себе. Хотя в начале у меня была такая идея... Но простая синхронизация ночного режима это как по мне лучшее решение.

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	reqform.info += "CONTENTS:<br>"
+	reqform.info += object.manifest
+	reqform.info += "<hr>"
+	reqform.info += "STAMP BELOW TO APPROVE THIS REQUISITION:<br>"
```
Вместо множественных += всё это можно внести в блок {" ... "}

Furukia replied:
Сделано

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+
+	reqform.update_icon()	//Fix for appearing blank when printed.
+
+	return reqform
```
Зачем нужно возвращать `reqform` если это никак не используется?

Furukia replied:
return убран

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	var/atom/Crate = new object.containertype(_loc)
+	Crate.name = "[object.containername] [comment ? "([comment])":"" ]"
+	if(object.access)
+		Crate:req_access = list(text2num(object.access))
```
Для чего здесь двоеточие? Зачем игнорирование рантайм эррора? Убрать.

Furukia replied:
Это и другие двоеточия как оказалось были важны и были скопированы с оригинального карго. Мне пришлось выкручиваться когда я их убирала. Не знаю нормален ли метод который я применила для этого или нет. Но теперь там нет двоеточий и вроде всё работает без рантаймов и ошибок.

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		return
+
+	//create the crate
+	var/atom/Crate = new object.containertype(_loc)
```
Название переменной должно быть с маленькой буквы

Furukia replied:
Сделано

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	slip.info +="Reason: [comment]<br>"
+	slip.info +="Supply Crate Type: [object.name]<br>"
+	slip.info +="[packagesAmt] PACKAGES IN THIS SHIPMENT<br>"
+	slip.info +="CONTENTS:<br><ul>"
```
Вместо множественных += всё это можно внести в блок {" ... "}

Furukia replied:
Сделано

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	slip.ordernumber = ordernum
+
+	var/stationName = "Syndicate RaMSS 'Taipan' Supply Mannifest"
+	var/packagesAmt = data_storage.shoppinglist.len + ((errors & MANIFEST_ERROR_COUNT) ? rand(1,2) : 0) // пометка ошибки возможной
```
"Пометка ошибки возможной"? Недописанная фраза или смысл в том, что тут возможна ошибка в количестве содержимого поставки?

Furukia replied:
Это я писала себе заметки когда искала причину бага. Забыла убрать эту заметку.

Furukia replied:
Заметка убрана дабы не смущать людей больше

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		var/datum/syndie_supply_packs/misc/randomised/SO = object
+		contains = list()
+		if(object.contains.len)
+			for(var/j=1, j<=SO.num_contained, j++)
```
```suggestion
		if(length(object.contains))
			for(var/j in 1 to SO.num_contained)
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		if(!typepath)	continue
+		var/atom/A = new typepath(Crate)
+		if(object.amount && A.vars.Find("amount") && A:amount)
+			A:amount = object.amount
```
Запрещенный символ в виде двоеточия, убрать

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		LC.manifest = slip
+		LC.update_icon()
+
+	return Crate
```
Для чего возврат, если он не используется?

Furukia replied:
Убран

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	linked_pads = list() // Обнуление на случай повторной синхронизации.
+	receiving_pads = list() // Мы же не хотим два одинаковых обьекта в одном списке
+	pads_cooldown = 0
+	for(var/obj/machinery/syndiepad/P in cargoarea)
```
Проходить по содержимому зоны - ужасно. В идеале постарайся придумать другой способ

Furukia replied:
Пока не сообразила метода получше...

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		telepads_status = "Pads ready"
+	else
+		to_chat(usr, "<span class='warning'>Synchronization failure! There's no pads in [cargoarea]!</span>")
+		telepads_status = "Pads not linked!"
```
Максимально неправильно написанная проверка на то, пустые ли листы. Я несколько минут пытался понять, что это вообще и для какой цели существует
```suggestion
	if (length(receiving_pads) && length(linked_pads))
		telepads_status = "Pads ready"
	else
		to_chat(usr, "<span class='warning'>Synchronization failure! There's no pads in [cargoarea]!</span>")
		telepads_status = "Pads not linked!"
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		to_chat(usr, "<span class='warning'>Synchronization failure! There's no pads in [cargoarea]!</span>")
+		telepads_status = "Pads not linked!"
+
+/obj/effect/abstract/syndie_data_storage/proc/cooldown()
+	if(is_cooldown)
+		telepads_status = "Pads on cooldown"
+		wait_time = round((last_teleport + pads_cooldown - world.time) / 10)
+		if(wait_time <=0)
+			wait_time = 0
+			telepads_status = "Pads ready"
+			is_cooldown = FALSE
+		return wait_time
```
А если не на кулдауне то вернёт что, null? А обработает ли код это?

Furukia replied:
Этот код вроде как в принципе не вызывается если мы не на кулдауне. Хотя я после целой ночи кодинга могу немного тупить. Позже перепроверю что да как тут

Vallat replied:
Всё еще актуален вопрос по поводу того, что при вызове функции нет возврата конкретного значения по умолчанию

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	for(var/typepath in subtypesof(/datum/syndie_supply_packs))
+		var/datum/syndie_supply_packs/SP = new typepath()
+		if(SP.name == "HEADER") continue		// To filter out group headers
+		syndie_supply_packs["[SP.type]"] = SP
```
```suggestion
	for(var/typepath in subtypesof(/datum/syndie_supply_packs))
		var/datum/syndie_supply_packs/SP = typepath
		if(initial(SP.name) == "HEADER")		// To filter out group headers
			continue
		syndie_supply_packs["[typepath ]"] = new SP
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+
+/obj/machinery/computer/syndie_supplycomp/proc/buy() // Этот код заточен под поиск точек для спавна, активации траты энергии и анимации падов
+
+	if(!data_storage.shoppinglist.len)
```
```suggestion
	if(!length(data_storage.shoppinglist))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+/obj/machinery/computer/syndie_supplycomp/proc/buy() // Этот код заточен под поиск точек для спавна, активации траты энергии и анимации падов
+
+	if(!data_storage.shoppinglist.len)
+		return 2
```
Зачем оно возвращает двойку?

Furukia replied:
Хер знает. Оно вроде и в обычном карго возвращало двойку. А я и не приметила. Убрала двойку

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	var/list/spawnTurfs = list()
+	var/list/recievingPads = data_storage.receiving_pads
+	for(var/j=1, j<=recievingPads.len, j++)
+		spawnTurfs += get_turf(recievingPads[j])
```
```suggestion
	for(var/j in 1 to length(recievingPads))
		spawnTurfs += get_turf(recievingPads[j])
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		for(var/i=1, i<=recievingPads.len, i++)
+			recievingPads[i].use_power(10000 / recievingPads[i].power_efficiency)
+			flick("sqpad-beam", recievingPads[i])
+			playsound(get_turf(recievingPads[i]), 'sound/weapons/emitter2.ogg', 25, 1, extrarange = 3, falloff = 5)
```
```suggestion
		for(var/i in 1 to length(recievingPads))
			recievingPads[i].use_power(10000 / recievingPads[i].power_efficiency)
			flick("sqpad-beam", recievingPads[i])
			playsound(get_turf(recievingPads[i]), 'sound/weapons/emitter2.ogg', 25, 1, extrarange = 3, falloff = 5)
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+
+	var/list/DSLP = data_storage.linked_pads
+
+	for(var/k = 1; k <= DSLP.len; k++)
```
```suggestion
	for(var/k in 1 to length(DSLP))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		for(var/atom/movable/MA in sellArea)
+			if(MA.anchored)
+				continue
+			if(istype(MA, /mob/dead) || istype(MA, /mob/living)) // Если окажется что на паде труп или человек, то это защитит его от уничтожения
```
`/mob/dead` это не труп, это обсервер.
`/mob/living` это не человек, это живое существо
Для проверки на то труп ли это есть переменная `stat` у всех живых мобов

Furukia replied:
Проверка stat у атома MA как оказалось дело тоже болезненное и пришлось искать пути как это сделать без всяких двоеточий. Вроде нашла.

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+				for(var/thing in MA)
+
+					// Sell manifests
+					data_storage.sold_atoms += " [thing:name]"
```
Убрать двоеточие и из-за этого переписать логику всего этого цикла

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+							var/denied = 0
+							for(var/i=1,i<=slip.stamped.len,i++)
+								if(slip.stamped[i] == /obj/item/stamp/denied)
+									denied = 1
```
```suggestion
						var/slip_stamped_len = length(slip.stamped)
						if(slip_stamped_len) //yes, the clown stamp will work. clown is the highest authority on the station, it makes sense
							// Did they mark it as erroneous?
							var/denied = 0
							for(var/i in 1 to slip_stamped_len)
								if(slip.stamped[i] == /obj/item/stamp/denied)
									denied = 1
									break
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+					// Sell exotic plants
+					if(istype(thing, /obj/item/seeds))
+						var/obj/item/seeds/S = thing
+						if(S.rarity == 0) // Mundane species
```
```suggestion
						if(!S.rarity) // Mundane species
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		SStgui.update_uis(src)
+		C.use(C.amount)
+	else
+		return ..()
```
```suggestion
/obj/machinery/computer/syndie_supplycomp/attackby(obj/item/I, mob/user, params)
	if(!powered())
		return 0
	if(istype(I, /obj/item/stack/spacecash))
		//consume the money
		var/obj/item/stack/spacecash/C = I
		playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 50, TRUE)
		data_storage.cash += C.amount
		to_chat(user, "<span class='info'>You insert [C] into [src].</span>")
		var/mob/living/carbon/human/H = user
		var/name = H.get_authentification_name()
		data_storage.blackmarket_message += "<span class='good'>+[C.amount]</span>: [name] adds credits to the console.<br>"
		SStgui.update_uis(src)
		C.use(C.amount)
		return 1
	return ..()
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		if(SO)
+			if(!SO.comment)
+				SO.comment = "No comment."
+			requests_list.Add(list(list("ordernum" = SO.ordernum, "supply_type" = SO.object.name, "orderedby" = SO.orderedby, "comment" = SO.comment, "command1" = list("confirmorder" = SO.ordernum), "command2" = list("rreq" = SO.ordernum))))
```
```suggestion
	for(var/datum/syndie_supply_order/SO as anything in data_storage.requestlist)
		if(!SO.comment)
			SO.comment = "No comment."
		requests_list += list(list("ordernum" = SO.ordernum, "supply_type" = SO.object.name, "orderedby" = SO.orderedby, "comment" = SO.comment, "command1" = list("confirmorder" = SO.ordernum), "command2" = list("rreq" = SO.ordernum)))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+	for(var/set_name in data_storage.shoppinglist)
+		var/datum/syndie_supply_order/SO = set_name
+		if(SO)
+			orders_list.Add(list(list("ordernum" = SO.ordernum, "supply_type" = SO.object.name, "orderedby" = SO.orderedby, "comment" = SO.comment)))
```
```suggestion
	for(var/datum/syndie_supply_order/SO as anything in data_storage.shoppinglist)
		orders_list += list(list("ordernum" = SO.ordernum, "supply_type" = SO.object.name, "orderedby" = SO.orderedby, "comment" = SO.comment))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+		if("teleport")
+			if(data_storage.telepads_status == "Pads not linked!"|| data_storage == null)
+				//Проверка на наличие хранилища данных
+				if(data_storage == null)
```
```suggestion
				if(!data_storage)
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+				if(data_storage == null)
+					compSync()
+				//Проверка на синхронизацию телепадов
+				if(data_storage.linked_pads == list())
```
Проверка на лист или проверка на пустой лист? В чем логика данного действия? Исходя из комментария, должна быть проверка на то, не пустой ли лист
```suggestion
				if(length(data_storage.linked_pads))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+				for(var/i = 1; i <= DSLP.len; i++)
+					DSLP[i].checks(usr)
+					data_storage.last_teleport = world.time
+					data_storage.is_cooldown = TRUE
```
```suggestion
				for(var/i in 1 tp length(DSLP))
					DSLP[i].checks(usr)
				data_storage.last_teleport = world.time
				data_storage.is_cooldown = TRUE
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+				idname = usr.real_name
+
+			//make our supply_order datums
+			for(var/i = 1; i <= amount; i++)
```
```suggestion
			for(var/i in 1 to amount)
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+			var/ordernum = text2num(params["ordernum"])
+			var/datum/syndie_supply_order/O
+			var/datum/syndie_supply_packs/P
+			for(var/i=1, i<=data_storage.requestlist.len, i++)
```
```suggestion
			for(var/i in 1 to length(data_storage.requestlist))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+
+		if("deny")
+			var/ordernum = text2num(params["ordernum"])
+			for(var/i=1, i<=data_storage.requestlist.len, i++)
```
```suggestion
			for(var/i in 1 to length(data_storage.requestlist))
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+				return
+			else
+				if(in_range(usr, src))
+					withdraw_cash(round(cash_sum), usr)
```
```suggestion
			if(cash_sum <= 0 || (!is_public && !is_authorized(usr)) || ..()))
				return
			withdraw_cash(round(cash_sum), usr)
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,699 @@
+
+/obj/machinery/computer/syndie_supplycomp/proc/withdraw_cash(cash_sum, mob/user)
+	if(cash_sum == null || cash_sum == 0)
+		return
```
Бесполезные проверки, такая ситуация не возникнет

Furukia replied:
Убрано

Vallat commented on code/game/machinery/portable_turret.dm:
```diff
@@ -582,6 +582,10 @@ GLOBAL_LIST_EMPTY(turret_icons)
+	var/obj/item/card/CID = L.get_id_card()
+	if(istype(CID, /obj/item/card/id/syndicate) && src.faction == "syndicate")
+		return TURRET_NOT_TARGET
+
```
```suggestion
	if(faction == "syndicate" && istype(L.get_id_card(), /obj/item/card/id/syndicate)
		return TURRET_NOT_TARGET

```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+	component_parts += new /obj/item/stock_parts/manipulator(null)
+	component_parts += new /obj/item/stack/cable_coil(null, 1)
+
+	RefreshParts()
```
```suggestion
/obj/machinery/syndiepad/Initialize()
	..()
	component_parts = list()
	component_parts += new /obj/item/circuitboard/quantumpad/syndiepad(null)
	component_parts += new /obj/item/stack/telecrystal(null)
	component_parts += new /obj/item/stock_parts/capacitor(null)
	component_parts += new /obj/item/stock_parts/manipulator(null)
	component_parts += new /obj/item/stack/cable_coil(null, 1)

	RefreshParts()
	return INITIALIZE_HINT_LATELOAD

/obj/machinery/syndiepad/LateInitialize()
	pad_sync()
```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+	if(console_link)
+		for(var/obj/effect/abstract/syndie_data_storage/S in myArea)
+			S.sync()
+			break
```
```suggestion
		var/obj/effect/abstract/syndie_data_storage/S = locate() in myArea
		S?.sync()
```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+				console_link = TRUE
+				for(var/obj/effect/abstract/syndie_data_storage/S in myArea)
+					S.sync()
+					break
```
```suggestion
		if(console_link)
			if(alert("Отвязать телепад от консоли?",, "Да", "Нет") == "Нет")
				return
			console_link = FALSE
			var/obj/effect/abstract/syndie_data_storage/S = locate() in myArea
			S?.sync()
		else if(!console_link)
			if(alert("Привязать телепад к консоли?",, "Да", "Нет") == "Нет")
				return
			console_link = TRUE
			var/obj/effect/abstract/syndie_data_storage/S = locate() in myArea
			S?.sync()
```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+	if(console_link)
+		to_chat(user, "<span class='notice'>Этот телепад привязан к консоли, воспользуйтесь ею для управления устройством!</span>")
+		return
+	if(receive == 0)
+		receive = 1
+		to_chat(user, "<span class='notice'>Включен режим получения посылок.</span>")
+		id = input("Задайте ID этому телепаду для получения им посылок")
+		linked_pad = null
+		target_id = null
+		return
+	if(receive == 1)
+		receive = 0
+		to_chat(user, "<span class='notice'>Включен режим отправки посылок.</span>")
+		target_id = input("Задайте ID телепада на который будут приходить посылки.")
+		linked_pad = null
+		return
```
```suggestion
	receive = !receive 
	if(receive)
		to_chat(user, "<span class='notice'>Включен режим получения посылок.</span>")
		var/new_id = input("Задайте ID этому телепаду для получения им посылок")
		if(new_id)
			id = new_id
		linked_pad = null
		target_id = null
	else
		to_chat(user, "<span class='notice'>Включен режим отправки посылок.</span>")
		var/new_target_id = input("Задайте ID телепада на который будут приходить посылки.")
		if(new_target_id && new_target_id != id)
			target_id = new_target_id
		linked_pad = null
```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+	default_deconstruction_screwdriver(user, "pad-idle-o", "sqpad-idle", I)
+
+/obj/machinery/syndiepad/proc/pad_sync()
+	for(var/obj/machinery/syndiepad/S in GLOB.machines)
+		if(src.console_link == TRUE && S.console_link == TRUE) //Мы не хотим привязываться к другим привязанным к консоли телепадам
+			continue
+		if(S.id == null)
+			continue
+		if(S.id == src.target_id)
+			if(S != src) //Так же мы не хотим привязываться сами к себе
+				linked_pad = S
+				break
```
```suggestion
	for(var/obj/machinery/syndiepad/S in GLOB.machines)
		if(S.console_link) //Мы не хотим привязываться к другим привязанным к консоли телепадам
			continue
		if(!S.id)
			continue
		if(S == src)
			continue
		if(S.id == target_id)
			linked_pad = S
			break
```

Furukia replied:
На 146-ой строчке стоит оставить проверку и на себя тоже. Ибо иначе с админского непривязанного к консоли телепада нельзя будет послать посылку в карго 
`if(src.console_link && S.console_link)` 
Важно чтобы он не привязывался к другим привязанным телепадам, только если он сам привязан.

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,278 @@
+					continue
+				// if dead body, never teleport. All my homies hate corpses, they stink!
+				if(istype(ROI, /mob/dead))
+					to_chat(user, "<span class='warning'>Error: '[ROI]' was not teleported! You cannot teleport dead bodies! It's gross!</span>")
```
Действительно ли мы хотим говорить что мы не смогли телепортировать **призрака**?

Furukia replied:
Убрано

Vallat commented on code/game/objects/effects/decals/syndie_logo_decals.dm:
```diff
@@ -0,0 +1,70 @@
+	icon_state = "logo8"
+
+/obj/effect/decal/syndie_logo/c
+
+/obj/effect/decal/syndie_logo/c/logo1
+	icon_state = "logo9"
+
+/obj/effect/decal/syndie_logo/c/logo2
+	icon_state = "logo10"
+
+/obj/effect/decal/syndie_logo/c/logo3
+	icon_state = "logo11"
+
+/obj/effect/decal/syndie_logo/c/logo4
+	icon_state = "logo12"
+
+/obj/effect/decal/syndie_logo/sc
+
+/obj/effect/decal/syndie_logo/sc/logo1
+	icon_state = "logo13"
+
+/obj/effect/decal/syndie_logo/sc/logo2
+	icon_state = "logo14"
+
+/obj/effect/decal/syndie_logo/sc/logo3
+	icon_state = "logo15"
+
+/obj/effect/decal/syndie_logo/s
+
+/obj/effect/decal/syndie_logo/s/logo1
+	icon_state = "logo16"
+
+/obj/effect/decal/syndie_logo/s/logo2
+	icon_state = "logo17"
+
+/obj/effect/decal/syndie_logo/s/logo3
+	icon_state = "logo18"
+
+/obj/effect/decal/syndie_logo/s/logo4
+	icon_state = "logo19"
```
**КОД ЭТО НЕ БАЗА ДАННЫХ**!!! Удалить всё к чертям, оставить только `/obj/effect/decal/syndie_logo` и РАССТАВЛЯТЬ НА КАРТЕ ЧЕРЕЗ ИНСТАНСЫ ОБЪЕКТА

Furukia replied:
Кхе кхе. Понято, принято, поправлено до 6 строчек. Сделано в карте через инстансы.

Vallat commented on code/game/objects/items/robot/robot_parts.dm:
```diff
@@ -264,7 +264,17 @@
+				aisync = FALSE
+				lawsync = FALSE
+
+			if(M.syndicate == 1)
```
```suggestion
			if(M.syndicate)
```

Vallat commented on code/game/objects/items/robot/robot_parts.dm:
```diff
@@ -356,7 +366,10 @@
-		forced_ai = select_active_ai(usr)
+		if(!sabotaged)
+			forced_ai = select_active_ai(usr)
+		if(sabotaged)
+			forced_ai = null
```
Не нужно

Furukia replied:
Убрано

Vallat commented on code/game/objects/items/stacks/sheets/mineral.dm:
```diff
@@ -363,6 +363,36 @@ GLOBAL_LIST_INIT(titanium_recipes, list(
 
 GLOBAL_LIST_INIT(plastitanium_recipes, list(
 	new/datum/stack_recipe("plas-titanium tile", /obj/item/stack/tile/mineral/plastitanium, 1, 4, 20),
+	//adding syndie crate recipe here - Furu
+	new/datum/stack_recipe("Suspicious crate", /obj/structure/closet/crate/syndicate, 10, 1, time = 20, one_per_turf = 1, on_floor = 1),
+
+	//adding syndie airlocks recipe's here - Furu
+	//WIP
+
+	new /datum/stack_recipe_list("Evil airlock assemblies", list(
+
+		new /datum/stack_recipe("Syndicate public airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_public, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate security airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_sec, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate cargo airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_cargo, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate atmospherics airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_atmos, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate research airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_research, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate medical airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_med, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate maintenance airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_maint, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate command airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_com, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate freezer airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_freezer, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate external maintenance airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_extmai, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("Syndicate engineering airlock assembly", /obj/structure/door_assembly/syndicate/door_assembly_syndie_engi, 4, time = 50, one_per_turf = 1, on_floor = 1),
+
+		//If i ever decide to make the rest syndie door types, here is the reference to what was not made - Furu
+		/*
+		new /datum/stack_recipe("standard airlock assembly", /obj/structure/door_assembly, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("science airlock assembly", /obj/structure/door_assembly/door_assembly_science, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("external airlock assembly", /obj/structure/door_assembly/door_assembly_ext, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("airtight hatch assembly", /obj/structure/door_assembly/door_assembly_hatch, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("maintenance hatch assembly", /obj/structure/door_assembly/door_assembly_mhatch, 4, time = 50, one_per_turf = 1, on_floor = 1),
+		new /datum/stack_recipe("multi-tile airlock assembly", /obj/structure/door_assembly/multi_tile, 8, time = 50, one_per_turf = 1, on_floor = 1),
+		*/
+	))
```
Давать возможность создавать синдидвери **всем** - плохая идея

Furukia replied:
Я считаю иначе. Эти двери прочнее всего на 50 единиц и для создания требуют пластитан, а не простой металл. Больше косметики для любителей строить штуки на обычной станции это не так уж и плохо учитывая, что двери в целом вообще никак кроме немного повышенной прочности на баланс и подобное не влияют. Это всё таки двери, а не какие-нибудь пушки.

Vallat commented on code/game/objects/structures/crates_lockers/crates.dm:
```diff
@@ -569,6 +569,29 @@
+	can_be_emaged = FALSE
+
+/obj/structure/closet/crate/secure/syndicate/emag_act(mob/user)
+	if(locked == 1 && broken == 0)
```
```suggestion
	if(locked && !broken)
```

Vallat commented on code/modules/mob/living/silicon/robot/robot.dm:
```diff
@@ -317,6 +317,8 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 		modules = force_modules.Copy()
 	if(mmi != null && mmi.alien)
 		modules = list("Hunter")
+	if(mmi != null && mmi.syndicate == 1)
```
```suggestion
	if(mmi?.syndicate)
```

Vallat commented on code/modules/mob/living/silicon/robot/robot.dm:
```diff
@@ -438,6 +440,18 @@ GLOBAL_LIST_INIT(robot_verbs_default, list(
 			icon_state = "xenoborg-state-a"
 			modtype = "Xeno-Hu"
 
+		if("Syndicate Saboteur")
+			spawn_syndicate_borgs(src, "Saboteur", get_turf(src))
+			qdel(src)
+
+		if("Syndicate Medical")
+			spawn_syndicate_borgs(src, "Medical", get_turf(src))
+			qdel(src)
+
+		if("Syndicate Bloodhound")
+			spawn_syndicate_borgs(src, "Bloodhound", get_turf(src))
+			qdel(src)
```
Weird flex but okay
```suggestion
		if("Syndicate Saboteur")
			spawn_syndicate_borgs(src, "Saboteur", get_turf(src))
			qdel(src)
			return

		if("Syndicate Medical")
			spawn_syndicate_borgs(src, "Medical", get_turf(src))
			qdel(src)
			return

		if("Syndicate Bloodhound")
			spawn_syndicate_borgs(src, "Bloodhound", get_turf(src))
			qdel(src)
			return
```

Vallat commented on code/modules/mob/living/simple_animal/bot/griefsky.dm:
```diff
@@ -241,3 +241,85 @@
+/mob/living/simple_animal/bot/secbot/griefsky/syndicate/sword_attack(mob/living/carbon/C)     // esword attack
+	src.do_attack_animation(C)
+	playsound(loc, 'sound/weapons/blade1.ogg', 50, 1, -1)
+	spawn(2)
```
Бесполезно без выделения области видимости под spawn() отдельным отступом, смысл не ясен. Удалить

Furukia replied:
spawn(2) это копипаст с обычного грифски. Вроде он и при удалении этой строчки прекрасно работает. Так что я удалила её у своей версии, но обычного не трогала.

Vallat commented on code/modules/mob/living/simple_animal/bot/griefsky.dm:
```diff
@@ -241,3 +241,85 @@
+		if((C.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+
+		threatlevel = C.assess_threat(src)
+
+
+		if(!("syndicate" in C.faction))
+			threatlevel = 20
+		else if("syndicate" in C.faction)
+			continue
+
+		var/obj/item/card/CID = C.get_id_card()
+		if(idcheck)
+			if(istype(CID, /obj/item/card/id/syndicate))
+				threatlevel = 0
```
```suggestion
		if(idcheck && istype(C.get_id_card(), /obj/item/card/id/syndicate))
			threatlevel = 0
		else if(!("syndicate" in C.faction))
			threatlevel = 20
		else
			threatlevel = C.assess_threat(src)
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+	unique_pet = TRUE
+	can_hide = 1
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/rouge/handle_automated_movement()
+	. = ..()
+	if(inventory_head)
+		regenerate_icons()
```
Зачем?

Furukia replied:
Остаток с того бага с летающими шляпами, не заметила и не удалила. Теперь удалено.

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+
+	var/on_CD = 0
+	act = lowertext(act)
+	switch(act)
+		if("hiss")
+			on_CD = handle_emote_CD()
+		else
+			on_CD = 0
+
+	if(!force && on_CD == 1)
+		return
```
```suggestion
	if(!force && act == "hiss" && handle_emote_CD())
		return
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+			regenerate_icons()
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/rouge/proc/shh(change, mob/M)
+	if(change)
+		if(change > 0)
+			if(M && stat != DEAD)
+				new /obj/effect/temp_visual/heart(loc)
+				custom_emote(1, "hisses happily!")
+		else
+			if(M && stat != DEAD) // Same check here, even though emote checks it as well (poor form to check it only in the help case)
+				custom_emote(1, "hisses angrily!")
```
```suggestion
	if(!M || stat)
		return
	if(change > 0)
		new /obj/effect/temp_visual/heart(loc)
		custom_emote(1, "hisses happily!")
	else
		custom_emote(1, "hisses angrily!")
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+	speak_emote = list("hisses")
+	emote_hear = list("Зевает", "Шипит", "Дурачится", "Толкается")
+	emote_see = list("Высовывает язык", "Кружится", "Трясёт хвостом")
+	set_light(0)
```
Зачем?

Furukia replied:
Зачем эмоции? Или зачем` set_light(0)`? Эмоции потому что надо! `set_light` потому что в оригинальном коде для корги, у корги есть риги и когда он их напяливает он начинает светиться, появляется бездыханность(ген) и защита от давления. А тут код это всё сбрасывает. У змеи конечно нет ригов. Эти куски кода в целом можно закомментировать пока змейке ещё и риг или что то подобное не подвезут. Пусть останутся на всякий.

Furukia replied:
`set_light(0)` и ещё пару строчек снизу были закомменчены поскольку пока не нужны, но я их оставила на всякий. Никогда не знаешь когда синдизмее подвезут риг или аналог.

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+	mutations.Remove(BREATHLESS)
+	minbodytemp = initial(minbodytemp)
+
+	if(inventory_head && inventory_head.snake_fashion)
```
```suggestion
	if(inventory_head?.snake_fashion)
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+		if(!SF.obj_color)
+			SF.obj_color = inventory_head.color
+
+		if(health <= 0 || src.rest) //труп или отдыхает
```
```suggestion
		if(stat || src.rest) //без сознания или отдыхает
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,296 @@
+
+		if(health <= 0 || src.rest) //труп или отдыхает
+			head_icon = SF.get_overlay()
+			if(health <= 0)
```
```suggestion
			if(stat)
```

Vallat commented on code/modules/ruins/syndicate_space_base.dm:
```diff
@@ -0,0 +1,231 @@
+/obj/effect/mob_spawn/human/space_base_syndicate/species_prompt()
+//Adding name and a gender pick. Furukai
+	if(allow_species_pick)
+		var/new_gender = input("Please select gender.") in list("Male","Female")
```
```suggestion
		var/new_gender = alert("Please select gender.",, "Male","Female")
```

Vallat commented on code/modules/ruins/syndicate_space_base.dm:
```diff
@@ -0,0 +1,231 @@
+			mob_gender = MALE
+		else
+			mob_gender = FEMALE
+		var/new_name = input("Enter your name:")
```
```suggestion
		var/new_name = input("Enter your name:") as text
```

Vallat commented on code/modules/ruins/syndicate_space_base.dm:
```diff
@@ -0,0 +1,231 @@
+		else
+			mob_gender = FEMALE
+		var/new_name = input("Enter your name:")
+		mob_name = new_name
```
```suggestion
		if(new_name)
			mob_name = new_name
```

Vallat commented on code/modules/ruins/syndicate_space_base.dm:
```diff
@@ -0,0 +1,231 @@
+				qdel(H.get_item_by_slot(slot_head))
+				H.equip_to_slot_or_del(new /obj/item/clothing/mask/gas/syndicate(H), slot_wear_mask)
+				H.equip_to_slot(new /obj/item/tank/plasma/plasmaman/belt/full(H), slot_l_hand)
+				H.equip_to_slot(I, slot_wear_id) // По непонятной мне причине другие методы считают что персонаж не может надеть предметы. Поэтому надеваем насильно!
```
Weird flex but okay

Furukia replied:
¯\_(ツ)_/¯

Vallat reviewed (changes requested):
Просмотреть все предложенные изменения, поменять многие названия переменных на более осмысленные (заменить всякие `k`, `j`, `m` на `iterator` или хотя бы `i`, расписать `amt` до `amount` и т.д., символы бесплатные). Пояснить многие моменты по типу непонятной логики и использования `:`

Furukia committed:
Apply suggestions from code review

Co-authored-by: Vallat <vox.crit2013@gmail.com>

Furukia committed:
Изменения и правки после ревью от Vallat-а

Изменения Кода:
* Теперь в карго в теории логируется, что было продано через investigate_log.
* Замечено и исправлено то что двери использовали оверлей старых дверей для фоток и бумажек, что к ним клеились. (Кто вообще придумал у дверей два вара с одинаковой ссылкой на оверлеи создать...)
* Добавлены проки которые возвращают обьект из атома и моба из атома соответственно. (Через них я чинила то что сломалось когда я убрала двоеточия. об этом ниже.)
* Зверушки из карго синдиката теперь когда спавнятся из коробки, спавнятся с faction "syndicate" Чтобы туррели по ним не стреляли через окна.

Следуя советам от Valat-а было изменено следующее:
* Инфо в манифестах и реквестах в карго теперь пишется в одной конструкции типа `{"..."}` вместо множества `+=`
* Были убраны все двоеточия что использовались вместо точки. Что всё к чертям сломало. Но я вроде починила... Вроде...
* Исправлены различные странности, опечатки и глупые мелочи. Например `return 2`
* Удалена куча бесполезных строчек с разными стейтами логотипа синдиката. Теперь этот код занимает всего 6 строчек. А на карте как уже писалось выше используются инстансы.

Furukia committed:
Исправление пола без кислорода в оружейке

* Заменила пол в оружейке ибо у него не было кислорода. упс.
* Ну и ещё пару закоменченых return-ов убрала

Furukia committed:
Замена пары полов на их аналог с включённой симуляцией

Vallat commented on code/__HELPERS/mobs.dm:
```diff
@@ -561,6 +561,20 @@ GLOBAL_LIST_INIT(do_after_once_tracker, list())
+		if(!.)
+			. = M
+		else
+			break
```
```suggestion
/proc/get_mob_in_atom_without_warning(atom/A)
	if(!istype(A))
		return null
	if(ismob(A))
		return A

	return locate(/mob) in A
```

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,702 @@
+			if(money2add > 0)
+				data_storage.blackmarket_message += "<span class='good'>+[money2add]</span>: We are pleased with your work. Here's your reward.<br>"
+			else if(money2add < 0)
+				data_storage.blackmarket_message += "<span class='bad'>[money2add]</span>: Don't anger us anymore! You won't be able to get away with such a little tax again.<br>"
```
Очень странный код. А если я добавлю -999999999? Должны ли быть какие-то рамки инпута и если да, то почему их нет?

Vallat commented on code/game/objects/objs.dm:
```diff
@@ -364,3 +364,17 @@ a {
 	// In the event that the object doesn't have an overriden version of this proc to do it, log a runtime so one can be added.
 	CRASH("Proc force_eject_occupant() is not overriden on a machine containing a mob.")
 
+/proc/get_obj_in_atom_without_warning(atom/A, mob/user = usr)
+	if(!istype(A))
+		return null
+	if(isobj(A))
+		return A
+
+	. = null
+	for(var/obj/O in A)
+		if(!.)
+			. = O
+		else
+			break
```
```suggestion
/proc/get_obj_in_atom_without_warning(atom/A)
	if(!istype(A))
		return null
	if(isobj(A))
		return A

	return locate(/obj) in A
```

Vallat commented on code/modules/mob/living/simple_animal/friendly/snake.dm:
```diff
@@ -59,3 +59,285 @@
+	popup.set_content(dat)
+	popup.open()
+
+/mob/living/simple_animal/hostile/retaliate/poison/snake/rouge/getarmor(def_zone, type)
+	var/armorval = 0
+
+	if(def_zone)
+		if(def_zone == "head")
+			if(inventory_head)
+				armorval = inventory_head.armor.getRating(type)
+		return armorval
+	else
+		if(inventory_head)
+			armorval += inventory_head.armor.getRating(type)
+	return armorval * 0.5
```
```suggestion
/mob/living/simple_animal/hostile/retaliate/poison/snake/rouge/getarmor(def_zone, type)
	var/armorval = inventory_head?.armor.getRating(type)
	if(!def_zone)
		armorval *= 0.5
	else if(def_zone != "head")
		armorval = 0
	return armorval 
```

Furukia committed:
Apply suggestions from code review

Co-authored-by: Vallat <vox.crit2013@gmail.com>

Furukia committed:
Ревёрт ломавшего карго изменения и мелкие правки

* Ревёрт одного изменения ломавшего карго
* Добавление return для cooldown-а
* Добавление комментария про админ кнопку.

Furukia committed:
Оптимизация логики синхронизации синдипадов с консолью

Теперь при инициализации все синдипады добавляются в глобальный список синдипадов. А при синхронизации с консолью вместо пробега через все объекты в зоне, мы пробегаем через этот лист.

Furukia committed:
Изменение хранилища данных синдикарго и другие мелочи

* Теперь хранилище данных это datum, а не эффект. В связи с чем логика некоторых вещей была переписана, но геймплейно ничего для карго не изменилось.
* Добавила в деструкторы различные ссылки сохранённые объектом которые надо удалять вместе с самими объектами.
* Всё же решилась добавить Офицеру телекоммов доступ к кабинету РД после действительно длительных размышлений на этот счёт...

Vallat commented on code/game/machinery/computer/syndie_cargo.dm:
```diff
@@ -0,0 +1,731 @@
+	sync()
+	orderNum = rand(1,9000)
+
+/datum/syndie_data_storage/Destroy(force)
+	if(force)
+		GLOB.data_storages -= src
+		linked_pads = null
+		receiving_pads = null
+		shoppinglist = null
+		syndie_supply_packs = null
+		discoveredPlants = null
+		techLevels = null
+		researchDesigns = null
+		..()
+		. = QDEL_HINT_HARDDEL_NOW
+	else
+		return QDEL_HINT_LETMELIVE
```
Так намного лучше
```suggestion
/datum/syndie_data_storage/Destroy(force)
	if(!force)
		return QDEL_HINT_LETMELIVE
	GLOB.data_storages -= src
	linked_pads = null
	receiving_pads = null
	shoppinglist = null
	syndie_supply_packs = null
	discoveredPlants = null
	techLevels = null
	researchDesigns = null
	return ..()
```

Vallat commented on code/game/machinery/syndiepad.dm:
```diff
@@ -0,0 +1,277 @@
+		S?.sync()
+
+/obj/machinery/syndiepad/proc/LocateDataStorage()
+	for(var/obj/machinery/computer/syndie_supplycomp/SC in GLOB.syndie_cargo_consoles)
+		if(get_area(SC) != get_area(src))
+			continue
```
```suggestion
	var/area/src_area = get_area(src)
	for(var/obj/machinery/computer/syndie_supplycomp/SC in GLOB.syndie_cargo_consoles)
		if(get_area(SC) != src_area)
			continue
```

Furukia committed:
Apply suggestions from code review

Co-authored-by: Vallat <vox.crit2013@gmail.com>

Furukia committed:
Добавила синди-рюкзаки

* Спрайты от ЭллГуда
* Рюкзаки спавнятся вместо обычных у сотрудников Тайпана
* Рюкзаки так же лежат в ящиках синдиката
* Рюкзаки имеют такую же вместительность и свойства как и обычные

Furukia committed:
Fixing merge conflict with tgui.bundle.js

Furukia committed:
Добавила Синди Пермиты и коробочку с ними

* Добавила синди версию construction permit-a
* Добавила коробочку с такими permit-ами к телекомщику и рд в ящики
* Добавила один такой permit в инженерку на столе.
* Спрайт для Пермита от ЭллГуда

Furukia committed:
Добавила звук при спавне на тайпане

* Добавила звук при спавне на тайпане как у контракторов при первом открытии аплинка.

BeebBeebBoob commented:
![image](https://user-images.githubusercontent.com/32985153/148499370-818fd9e0-bfd7-4f4c-906f-923eae3e0e9f.png)
Прям по непонятной причине тут бьёт током если идти на стенку.

Den109G commented:
Это из-за фулл-тайл окон. Почему-то, игра не думает что они есть, и просто бьёт, из-за решётки под напряжением.

Den109G commented:
Дополню сообщение выше.
![image](https://user-images.githubusercontent.com/87372121/148512001-e396d514-c1d0-48dc-8c6e-256d8a7d8214.png)
И тут в ботанике бьёшься током. Предположу, что проблема не в карте, а в окнах.

BeebBeebBoob commented:
![image](https://user-images.githubusercontent.com/32985153/148586034-34fcc11f-2701-4693-a86f-786c72661c2f.png)
Уже пошли мемы про синдибазу и их приколов. Кстати вирусология как я понял никому не доступна! Даже учёным а это.. обидно и плохо.

carlsmei commented:
**Баг**

Хочешь поднять кредитов в карго? Тебе нужны хотя бы 3000 чёртовых кредитов. Ладно, ближе к сути, чтобы получать большое количество **ДЕНЯК**, можно просто покупать ящик плазмы и, надо же, отправлять обратно. Профит неплохой выходит.

![изображение](https://user-images.githubusercontent.com/47307437/148632169-a9cc74b9-28b5-4ad4-8c91-33332dcffe6a.png)

carlsmei commented:
В добавок к верхнему сообщению от меня

Деньги также можно получать через, КАЗАЛОСЬ БЫ, сервисного борга, у которого есть возможность печатать 10 кредитов буквально из жопы.

Так выглядит майнинг-станция по добыче кредитов
![изображение](https://goose.durka.xyz/f/cc2c8717-a23b-4b1c-99d8-844ca8c03cbc.png)

![изображение](https://user-images.githubusercontent.com/47307437/148633275-7c12246e-4e6d-43cc-9628-9c7fa5529d7e.png)

BeebBeebBoob commented:
> Деньги также можно получать через, КАЗАЛОСЬ БЫ, сервисного борга, у которого есть возможность печатать 10 кредитов буквально из жопы.
>

Это вообще пойдёт под другой ПР и это будет БОЛЬШОЙ ПР ибо это надо целую экономику менять. На самом парадизе экономика просто смешно. Часть либо бесплатное часть платное. У нас полу-коммунизм так сказать. На самом оффе в гитхаб тупо говорят что система экономики самый старый код и самый ужасный.

BeebBeebBoob commented:
https://github.com/ss220-space/Paradise/blob/d93505df67673cb69fc8c311743a8cd2bf38607e/code/datums/syndiesupplypacks.dm#L1302-L1307

тут лучше тогда поменять на 30000 ибо плазма штука стоит 500 и просто выходит что ящик стоит 6 плазмы.

AugustBebel commented:
ТГ придэ, порядок наведэ. Фиксить парадизы на лету это сизифов труд.

Furukia commented:
> **Баг**
> 
> Хочешь поднять кредитов в карго? Тебе нужны хотя бы 3000 чёртовых кредитов. Ладно, ближе к сути, чтобы получать большое количество **ДЕНЯК**, можно просто покупать ящик плазмы и, надо же, отправлять обратно. Профит неплохой выходит.
> 
> ![изображение](https://user-images.githubusercontent.com/47307437/148632169-a9cc74b9-28b5-4ad4-8c91-33332dcffe6a.png)

Ля. Я же вроде специально смотрела чтобы этого не случилось. Как же я так в расчётах ошиблась...
![7221-menhera-embarassed](https://user-images.githubusercontent.com/67807868/148650598-8fbaddea-3074-4b0b-a50c-00809cd46430.png)

Furukia committed:
Изменение способа спавна тайпана

* Теперь Тайпан спавнится всегда на 4 Z - уровне.
* На него всё ещё можно попасть через путешествие в космосе, но теперь он не вызывает 100000 пробегов по лупу в попытке себя разместить, а просто спавнится на выделенном уровне.
* Так же сделала пару изменений на самой карте и в расположении астероидов и туррелей

Furukia committed:
Фиксы абузов и багов

* Теперь цена плазмы в карго не позволяет зарабатывать на её перепродаже
* Теперь в карго есть ящик с титаном
* Теперь синди пермиты создают синдизоны вместо простых
* Консоль рнд при постройке в синдизоне привяжется к синди серверу. Больше не будут воровать у станции науку
* Построенная на синдизоне печка для ресурсов теперь автоматически настраивается на базовый синдидоступ

Furukia committed:
Добавила особую частоту для тайпана

* Добавила особую радио частоту для Тайпана, особый наушник и ключи шифрования.
* Для общения на частоте используется ключ ":," или ":б"
* Агент телекомов способен общаться по частоте тайпана, трейторов и обычной нтшной.
* Остальные способны общаться только на частоте Тайпана, но ничего не запрещает им создать ключи шифрования и общаться и на трейторской линии. Такой просто позволит убавить информационный шум для обычных трейторов или нюки.

Furukia committed:
Добавила роль медика и ботаника. Убрала две роли учёных

* Вместе с ролями поставляется два новых доступа в мед и ботанику.
* У повара всё так же есть доступ в ботанику
* У учёных всё так же есть доступ в мед зоны. Доступ в главный зал мед отдела общий.
* Вместе с двумя профессиями и доступами так же были добавлены новые карты и доступ добавлен в основные proc-и что пользуются ими.

Furukia committed:
Update misc.dm

Den109G commented:
Повар всё ещё спавниться с прошлым наушником. Немного неприятно, но не критично.
И да, раз появились новые ключи для Тайпана, то может заменить синди-ключи в шкафах на новые?

Furukia committed:
Фиксы абузов и багов 2

* Увеличение цены ящика "Imported Vending Machines" до 60 кредитов, для предотвращения возможных абузов с перепродажей.
* Лестница Иакова больше не будет попадать на синдибазу
* Грифски теперь говорит на частоте Тайпана
* Повару выдан правильный наушник
* Ночное освещение тайпана, больше не зависит от кода на станции
* Добавления частоты тайпана админ офицеру синдиката и боргам синдиката что были построены на тайпане.
* Борги собранные из Емагнутого заранее корпуса больше не имеют встроенной камеры.
* Заменила старые ключи шифрования в ящиках на Тайпанские
* Уменьшила начальный заряд смесов на тайпане
* Добавила запасную карту офицера телекомов
* Добавила доступ медика на клетки мартышек в вирусологии

Furukia committed:
Синди РЦД и Плата с для Аирлоков.

* Добавила синди рцд. Оно ставит стены и окна из пластитана. Синдидвери с синдидоступами и ломает укреплённые стены.
* Добавила рецепт для синди рцд в протолат в раздел нелегала
* Добавила рецепт картриджей для синди рцд в протолат в раздел нелегала
* Добавила плату с синди доступами во взломанный автолат

PunishedHideo commented:
Что-то на чёрном рынке оружие никакое не продаётся, кредиты не дают, странный какой то чёрный рынок...
Функция продажи оружия там вообще есть?

Den109G commented:
Нету функции продажи оружия, как и у обычного карго.
![image](https://user-images.githubusercontent.com/87372121/149290084-99bce629-0363-406c-96d6-21b3ec39c7be.png)
![image](https://user-images.githubusercontent.com/87372121/149290182-a5bd11cf-60b6-4e05-91e9-b3248ba7aa9e.png)
Карго синди принимает только: плазму, секретные документы НТ, технологии, семена растений, дизайны, манифесты от заказов и ящики.

Furukia committed:
Фиксы абузов и багов 3

* Добавила стандартный синдидоступ в функцию собирающую доступы для рцд, платы шлюзов и т.д.
* Собранные на тайпане аир алярмы теперь имеют синдидоступ
* Собранная на тайпане клонёрка теперь имеет синдидоступ и не орёт в канал медиков. Раундстартовая тайпановская тоже.
* Собранные на тайпане АПК теперь имеют синдидоступ
* Ушедшие в тайпанское крио (красное) более не фиксируются в общем канале связи
* При разборе Тайпанского аирлока теперь выпадает плата с Тайпанскими доступами
* При отсутствии введённого имени когда заходишь на синдирольку теперь рандомится имя. (Больше никаких людей с именем Syndicate Scientist)
* Серверная синдибазы больше не выдаёт аир алярмы в инженерке НТ
* Теперь Синди рцд может использовать для зарядки только свои красные картриджи с материей. А обычное рцд только свои синие.
* Ещё я заменила текстурку Синди РЦД ибо совершенно забыла сразу поменять спрайт на более новый от ЭлГуда. (Упс)
* Переименовала пару криво названных труб
* Немного переделала начальный ТЭГ и расстановку в атмосе

Furukia committed:
Фиксы абузов и багов 4

* Изменила принцип работы синдикат телепортера. Теперь при попытке выбора места телепорта он блокирует все кнопки на 30 секунд пока не будет выбрана зона телепорта. Это защищает от создания сразу кучи порталов и поломок системы из-за этого. Так же теперь нельзя активировать телепорт стоя далеко от консоли или не имея доступа телекомщика.
* Теперь синдипады способны телепортировать мехов. Они всё ещё не будут телепортировать если в них сидит человек и на синдипаде при этом не выставлен админский флаг allow_humans
* Теперь в логах при заходе за гост роль так же пишется профессия этой роли если она имеется
* Теперь синдиборги всегда имеют синдиробомозг, даже если они были куплены нюкой
* Теперь сервера и консоль серверов собранные на тайпане всегда будут трейторскими
* Теперь при нажатии Cansel при выборе расе для спавна, спавн будет отменён.

Furukia committed:
Изменения контента и карты

* Полностью переделала телепортную зону добавив туда так же два синдипада для прибытия подмоги на Тайпан от синди ЦК и в целом расширив её потому, что раньше когда таки завязывались сражения с использованием телепортов, все кучковались в маленькой комнатке. Теперь у вас будет пространство для манёвров и не только.
* На Z2 создала севернее комнатку для телепортера и синдипадов через которые можно отправлять поддержку на Тайпан.
* Добавила холодильникам на Тайпане новый спрайт от ЭллГуда, а так же холодильник для слаймов построенный или поставленный на Тайпане всегда будет иметь доступ для учёных.
* Покрасила в чёрный единственную клетку зарядки для мехов на Тайпане.

Den109G commented:
Думаю стоит поменять ПДА у РД и Офицера. Точнее - убрать картридж.
Сейчас и так много грифозных вещей можно сделать на Тайпане, даже взорвать ПДА почти раундстартом. Это не очень классно, как по мне. Начинается раунд, а ты уже без ноги и уже стоит Красный Код.
Само предложение -
Добавить в `pdas.dm` эти строчки и заменить ПДА в шкафчиках:
```
/obj/item/pda/syndicate/(nobomb, к примеру)
	default_cartridge = null
```
Или
```
/obj/item/pda/taipan
	name = "Secret PDA"
	icon_state = "pda-syndi"
	owner = "Дядя Петя"
	
obj/item/pda/taipan/New()
	..()
	var/datum/data/pda/app/messenger/M = find_program(/datum/data/pda/app/messenger)
	if(M)
		M.m_hidden = 1
```

Furukia committed:
Фиксы абузов и багов 5

* Починила доступ к телепортеру (Криво прописала в коде в прошлый раз)
* Убрала детоматрикс картридж из синдипдашек тайпана и переименовала их

Furukia committed:
Новые спрайты

* У гранатомётов снаружи тайпана теперь уникальный спрайт от ЭллГуда
* У криокапсул так же новый рескин от ЭллГуда
* У имплантера так же новый скин от ЭллГуда

Furukia committed:
Временное удаление

Furukia committed:
Возвращение спрайтов

Furukia committed:
Фиксы абузов и багов 6

* Изменила принцип работы наушников тайпана. Теперь без ключа шифрования тайпана, нельзя слышать Тайпановскую частоту (жёлтенькую). Даже трейторам с покупным ключом. (Спасибо @semoro за помощь с кодом)
* Теперь при применении тайпанского наушника, название частоты тайпана в TGUI подсвечено его цветом.

Furukia committed:
Худы, спрайты ПДА, крио

* Добавила больше огнетушителей на Тайпан
* Добавила Тайпановский худ для каждой профессии. Спрайты от ЭлГууда.
* Добавила Админский вызов в VV-шку, для удобной выдачи этого худа в случае чего.
* Теперь после ухода тайпановца в крио. Крио заменяется на новый спавнер, для занятой им ранее роли.
* Новые спрайты для ПДА у РД и Телекомщика от ЭллГуда

Furukia committed:
Фиксы абузов и багов 7

* За продажу дисков в синдикарго задание на продажу технологий обычных НТ учёных больше не выполняется.
* Теперь криоподы проверяют какого они типа и не спавнят новый спавнер если тайпановец уходит в обычное крио
* Теперь админская кнопка cryo стала умнее и кидает обычных людей в обычное крио, а тайпановцев в тайпанское
* Починили синди и тайпан радио частоты, снова, вроде...
* Теперь фабрикаторы подов/мехов при постройке на тайпане требуют синдидоступ.

Furukia committed:
Нововведения и другие изменения

* Теперь на ЦК и в товарах сол торговца спавнится автоимплантер со старым скином
* Теперь шатлы ССТ и СИТ могут пристыковаться около тайпана
* Теперь у телепортера на тайпане есть плата. А значит при поломке его можно починить, так же как и собрать новый. А ещё у него теперь полный резист к пулям. (Чтобы туррели не ломали)
* Добавила в карго в раздел РНД ящик с платой телепортера синдиката с уникальным скином.

Furukia committed:
Update tgui.bundle.js

Furukia committed:
Фиксы абузов и багов 8

* Добавила в карты ССТ фул доступ на тайпане
* Добавила возможность синдиучёным настраивать Хим-раздатчик на ботанический
* Теперь абилка малф ИИ не будет взрывать РЦД на тайпане и админском уровне.
* Теперь лут в техах спавнится в обычных или синдиящиках вместо шкафов с синдиастероида с нипами. Сам астероид остался неизменённым.
* Изменила туррели снаружи тайпана создав вместо них спавнеры:
	** Позиций для туррелей стало в разы больше
	** Теперь туррели имеют шанс в 66% для гранатомётов и в 50% для пулемётов на спавн. (Раньше спавнились всегда)
	** Починила и переместила АПК для наружных туррелей в космос под кабинетом РД
	** Теперь туррели пассивно тратят энергию с тайпана.

Furukia committed:
Бюрократия. Синдикратия.

* Теперь можно поставить логотип синдиката на бумаге ручкой написав [slogo]
* Добавлена печать для тайпана. Спрайт самой печати на листке от wolferokt
* Добавлен синдипринтер с уникальным спрайтом
* Добавлены несколько синдикатских форм для любителей бюрократии на телекомщике/рд. Сами формы:
	** Официальное письмо (Админская форма, доступна для печати только синдиофицерам)
	** Запрос особой доставки
	** Экстренное письмо
	** Отчёт о ситуации на станции
	** Отчёт о разработке вируса
	** Список будет пополняться...

Den109G commented:
Предложение по фиксу РнД Тайпана -
Поменять в rdconsole.dm в эти строки
```
...
	if(is_taipan(z))
		syndicate = 1
		req_access = list(ACCESS_SYNDICATE_SCIENTIST)
...
```
Значение, что у сервера Тайпана (23 или 0027). То есть вот так должен выглядеть код, что подключит консоль к серверу Тайпана:
```
if(is_taipan(z))
		syndicate = 1
		req_access = list(ACCESS_SYNDICATE_SCIENTIST)
		id = 23
```

Если нужно чтобы Тайпан мог скачивать данные НТ - Нужно добавить в лист ID серверов 1 (консоль РнД НТ)
![image](https://user-images.githubusercontent.com/87372121/154429198-b25522b6-c09d-44f4-9d82-932b3838e979.png)
(0027 превращается в игре в 23, так что маленькие изменения)

Эти изменения разделят РнД Тайпана и НТ (частично)

Furukia committed:
Ввод ограничений по времени для захода на тайпан

Учёный, медик, повар, ботаник, карготех - 10 часов живым
Инженер - 10 часов в инженерном отделе
РД - 50 часов в научном отделе
Офицер Телекоммуникаций - 50 часов за командование

Furukia committed:
Фиксы абузов и багов 9

* Заменила маленькие вентиляторы на декали со скином вентилятора. (Чтобы они не блочили проход атмосу)
* Нашла злополучную уязвимость в рнд и добавила необходимые проверки для запрета скачивания/загрузки не на те сервера.
* Ещё немного поправила код телепортера. Сменила скин на Красный синдикатовский. Добавила логирование применения телепорта. Так же теперь порталы существуют около 30 секунд, затем автоматически закрываются. Это не касается телепортов на z2. Они как и раньше вечные.
* Поправила код проверки целей у турелей. Так же НТшные борги созданные на Тайпане, теперь не будут получать пули в окуляры.

Furukia committed:
Синди мехи и другие нововведения

* Добавлены Синди-Мехи! Дарк-Гигакс и совершенно новая вариация дюранда от синдиката - Ровер
* Добавлен синди фабрикатор для мехов! (Плата в карго)
* Удалён Синди-Робомозг из карго. Теперь он делается в фабрикаторе.
* Добавлены уникальные скины для стандартных рнд машин, а так же синдикатский стиль для UI главной РнД Консоли

* Дарк-Гигакс такой же по характеристикам как и раньше, но теперь его можно собрать. Способ сборки не отличается от стандартного Гигакса, за исключением деталей из Синди-Фабрикатора
* Ровер имеет следующие показатели защиты: melee = 30, bullet = 15, laser = 50, energy = 50, bomb = 20, bio = 0, rad = 50, fire = 100, acid = 100. Способ сборки не отличается от стандартного Дюранда за исключением деталей из Синди-Фабрикатора. Вместо стандартной способности дюранда, способен создавать Энергетическую стену в 3 тайла шириной по аналогии со способностью мага. Через стену могут проходить только Синдикатовцы.
* Оба меха имеют двигатели для полётов в космосе.

Furukia committed:
Фиксы абузов и багов 10 + пару новых мелочей

* Вентилятор в прошлом не пропускавший воздух в мусоросбросе теперь снова его не пропускает, остальные вентиляторы всё так же декали.
* Переделан выход в космос в инженерном отделе и починены доступы на нём.
* Клонированные на тайпане больше не теряют флаг syndicate защищающий от турелей
* Исправлена ошибка отображения деструктора возникавшая при доставании их него предмета без разборки. (Скин менялся на НТшный)

# Нововведения

* Генерал синди приобрёл уникальный скин от ElGood-а
* Добавлен худ для построенных на тайпане боргов.
* Вместо заранее заготовленных потайных стен на тайпане теперь полно спавнеров которые имеют шанс заспавнить потайную или обычную стену на своей клетке. Пару важный стен были не тронуты.
* Добавлен Пакман в инженерный отдел.
* Вместо двух канистр с плазмой в атмосе теперь стоит 4 не прикрученных. Так же на столе была добавлена бумажка с инструкцией по запуску ТЭГ-а.

Furukia committed:
Обход ограничения на часы для модеров и админов

Делаю отдельным коммитом в связи с тем, что без бд мне нормально не проверить его работоспособность. Поэтому в случае если он не работает или вызовет какие то проблемы, коммит можно будет легко откатить. По крайней мере в моих условиях код проблем не вызвал. Воть

Furukia committed:
Resolving merge conflicts after rebase

Furukia committed:
Мелкие фиксы/изменения тайпана

* Правки проблем с доступами в вирусологии
* Добавление упущенных скинов smartfridge
* Добавление принтера в вирусологию
